File: Project Templates/VIPER Single View App.xctemplate/Objective-C/FirstRouter.m
//___FILEHEADER___
 
#import "FirstRouter.h"
#import "FirstVC.h"
#import "FirstInteractor.h"
#import "FirstPresenter.h"
 
@implementation FirstRouter

+ (UIViewController *)createModule
{
    
    /**
     * Berfungsi untuk menghubungkan komponen VIPER per Modul
     */
    
    FirstVC *viewController = [self.mainStoryboard instantiateViewControllerWithIdentifier:@"FirstVC"];
    FirstInteractor *interactor = [[FirstInteractor alloc] init];
    FirstRouter *router = [[FirstRouter alloc] init];
    FirstPresenter *presenter = [[FirstPresenter alloc] initWithInterface:viewController interactor:interactor router:router];
    
    /// Menghubungkan
    viewController.presenter = presenter;
    router.viewController = viewController;
    
    return viewController;
}

+ (UIStoryboard *)mainStoryboard
{
    
    /**
     * Berfungsi untuk mengembalikan nilai storyboard
     * Untuk nama storyboard, dikondisikan
     */
    
    return [UIStoryboard storyboardWithName:@"Main" bundle:NSBundle.mainBundle];
}
 
@end

File: Project Templates/VIPER Single View App.xctemplate/Swift/FirstRouter.swift
//___FILEHEADER___

import UIKit

class FirstRouter: FirstRouteProtocol {

    class func createModule() -> UIViewController{

        /**
         * Berfungsi untuk menghubungkan komponen VIPER per Modul
         */

        let view = mainstoryboard.instantiateViewController(withIdentifier: "FirstVC") as! FirstVC
        let presenter: FirstPresenterProtocol & FirstInteractorOutputProtocol = FirstPresenter()
        let interactor: FirstInteractorInputProtocol = FirstInteractor()
        let router: FirstRouteProtocol = FirstRouter()

        /// Conecting
        view.presenter = presenter
        presenter.view = view
        presenter.wireFrame = router
        presenter.interactor = interactor
        interactor.presenter = presenter

        return view
    }

    static var mainstoryboard: UIStoryboard {

        /**
         * Berfungsi untuk mengembalikan nilai storyboard
         * Untuk nama storyboard, dikondisikan
         */

        return UIStoryboard(name:"Main",bundle: Bundle.main)
    }
    
}

